vortex-file/src/read/layouts/inline_dtype.rs

```diff
@@ -125,19 +126,19 @@ impl LayoutReader for InlineDTypeLayoutReader {
         self.child_reader()?.add_splits(row_offset, splits)
     }
 
-    fn read_selection(&self, selector: &RowMask) -> VortexResult<Option<BatchRead>> {
+    fn poll_read(&self, selector: &RowMask) -> VortexResult<PollRead<Option<ArrayData>>> {
```

Comment:
I like this a lot better, but maybe we could typedef an `ArrayDataRead` and a `MetadataRead`?